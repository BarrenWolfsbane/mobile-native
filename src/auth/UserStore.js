import {
  observable,
  action
} from 'mobx';

import channelService from './../channel/ChannelService';
import sessionService from '../common/services/session.service';
import UserModel from './../channel/UserModel';
import { MINDS_FEATURES } from '../config/Config';
import { Alert } from 'react-native';

/**
 * Login Store
 */
class UserStore {
  @observable me = {};

  @action
  setUser(user) {
    if (!user || !user.guid)
      return
    this.me = UserModel.create(user);
  }

  @action
  setRewards(value) {
    this.me.rewards = !!value;
  }

  @action
  setWallet(value) {
    this.me.eth_wallet = value;
  }

  @action
  setTosLastUpdate(value) {
    this.me.last_accepted_tos = value;
  }

  hasRewards() {
    return !!(this.me && this.me.rewards);
  }

  hasEthWallet() {
    return !!(this.me && this.me.eth_wallet);
  }

  hasAvatar() {
    return this.me.icontime !== this.me.time_created;
  }

  @action
  clearUser() {
    this.me = {};
  }

  @action
  async load(refresh = false) {
    if (!refresh) this.me = {};

    let response = await channelService.load('me');

    //if (!response.channel) {
    //  return sessionService.setToken(null);
    //}

    this.setUser(response.channel);
    if (this.me.canCrypto) {
      MINDS_FEATURES.crypto = true;
    }
    return this.me;

  }

  isAdmin() {
    return this.me.admin;
  }

  @action
  reset() {
    this.me = {};
  }

}

export default UserStore;
