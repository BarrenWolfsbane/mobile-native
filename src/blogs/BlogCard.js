import React, {
  Component
} from 'react';

import {
  Text,
  Image,
  View,
  Button,
  StyleSheet,
} from 'react-native';

import {
  observer,
  inject
} from 'mobx-react/native'

import Icon from 'react-native-vector-icons/Ionicons';
import { MINDS_CDN_URI } from '../config/Config';
import FastImage from 'react-native-fast-image';
import formatDate from '../common/helpers/date';

/**
 * Blog Card
 */
@observer
export default class BlogCard extends Component {

  /**
   * Get Blog Avatar
   */
  getAvatar() {
    const channel = this.props.entity.ownerObj;
    return { uri: MINDS_CDN_URI + 'icon/' + channel.guid + '/small/' + channel.icontime };
  }

  /**
   * Render Card
   */
  render() {
    const blog = this.props.entity;

    const image = { uri: blog.thumbnail_src };

    return (
      <View>
        <FastImage source={image} style={styles.banner} resizeMode={FastImage.resizeMode.cover} />
        <View style={styles.headertextcontainer}>
          <View style={styles.namecontainer}>
            <View style={styles.namecol}>
              <Text style={styles.name}>{blog.title}</Text>
              <View style={styles.ownerContainer}>
                <Image source={this.getAvatar()} style={styles.avatar} />
                <Text style={styles.username}>{blog.ownerObj.username.toUpperCase()}</Text>
                <Text style={styles.createdDate}>{formatDate(blog.time_created)}</Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    )
  }
}


const styles = StyleSheet.create({
  headertextcontainer: {
    padding: 8,
    paddingLeft: 15,
    paddingRight: 15,
    alignItems: 'stretch',
    flexDirection: 'column',
    width: '100%',
  },
  namecol: {
    flex:1
  },
  namecontainer: {
    flexDirection: 'row',
    flex:1,
  },
  ownerContainer: {
    flex:1,
    flexDirection: 'row',
    alignSelf: 'flex-start',
    padding:3 ,
    alignItems: 'center',
    justifyContent: 'center'
  },
  banner: {
    flexDirection: 'row',
    alignItems: 'stretch',
    width: '100%',
    height: 150,
  },
  headercontainer: {
    flex: 1,
    height: 100,
    flexDirection: 'row',
  },
  username: {
    paddingLeft: 5,
    fontSize: 10,
    color: '#999'
  },
  name: {
    fontWeight: 'bold',
    fontSize: 16,
  },
  createdDate: {
    paddingLeft: 5,
    fontSize: 8,
  },
  countercontainer: {
    paddingLeft: 130,
    height: 60,
    flexDirection: 'row'
  },
  avatar: {
    height: 20,
    width: 20,
    borderRadius: 10
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#FFF'
  }
});
