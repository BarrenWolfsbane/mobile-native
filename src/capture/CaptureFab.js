import React, { Component } from 'react';

import {
  observer,
} from 'mobx-react/native'

import {
  View,
  StyleSheet
} from 'react-native';

import { Icon } from 'react-native-elements'
import testID from '../common/helpers/testID';
import settingsService from '../settings/SettingsService'
import settingsStore from '../settings/SettingsStore'
import featuresService from '../common/services/features.service';

@observer
export default class CaptureFab extends Component {

  /**
   * Nav to activity full screen
   */
  navToCapture = () => {
    const { state } = this.props.navigation
    this.props.navigation.navigate('Capture', {group: this.props.group, parentKey: state.key});
  }

  render() {
    if (featuresService.has('navigation-2020')) {
      return null;
    }
    
    return (
      <Icon
        raised
        reverse
        name="md-create"
        type='ionicon'
        color='#4690DF'
        size={28}
        containerStyle={ settingsStore.leftHanded ? styles.leftSide : styles.rightSide }
        onPress={() => this.navToCapture()}
        testID={this.props.testID}
        />

    );
  }
}

const styles = StyleSheet.create({
  rightSide: {
    position:'absolute',
    // backgroundColor:'#4690DF',
    bottom:8,
    zIndex:1000,
    right:8
  },
  leftSide: {
    position:'absolute',
    // backgroundColor:'#4690DF',
    bottom:8,
    zIndex:1000,
    left:8
  }
});
