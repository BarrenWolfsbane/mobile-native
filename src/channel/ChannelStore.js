import {
  observable,
  action
} from 'mobx';

import channelService from './ChannelService';

/**
 * Channel Store
 */
class ChannelStore {
  @observable channel = {};

  @action
  setChannel(ch) {
    this.channel = ch;
  }

  @action
  clearChannel() {
    this.channel = {};
  }

  @action
  load(guid) {
    this.channel = {};
    channelService.load(guid)
      .then(action(response => {
        console.log(response);
        this.channel = response.channel;
      }))
      .catch(err => {
        console.log('error', err);
      });
  }
}

export default new ChannelStore();
