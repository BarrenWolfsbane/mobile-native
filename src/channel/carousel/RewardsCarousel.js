import React, {
  Component
} from 'react';

import {
  Text,
  View,
  Dimensions
} from 'react-native';

import {
  observer,
} from 'mobx-react/native'

import Icon from 'react-native-vector-icons/Ionicons';
import Carousel from '../../common/components/Carousel';

/**
 * Channel Rewards Carousel
 */
@observer
export default class RewardsCarousel extends Component {

  state = {
    width: Dimensions.get('window').width
  }

  /**
   * Handle dimensions changes
   */
  dimensionChange = () => {
    this.setState({
      width: Dimensions.get('window').width
    });
  }

  componentWillMount() {
    Dimensions.addEventListener("change", this.dimensionChange);
  }

  componentWillUnmount() {
    // allways remove listeners on unmount
    Dimensions.removeEventListener("change", this.dimensionChange);
  }

  /**
   * Render component
   */
  render() {
    const rewards = this.props.rewards;
    const styles  = this.props.styles;

    const rewardsArray = [];

    if (rewards.money) {
      rewards.money.map((rew) => {
        rewardsArray.push(
          <View key={`rewards${rew.amount}`} style={styles.carouselitems}>
            <View>
              <Text style={styles.rewardamount}>${rew.amount}+</Text>
              <Text style={styles.rewarddesc}>{rew.description}</Text>
            </View>
            <Icon style={styles.rewardicon} name={'ios-flash'} size={36} />
          </View>
        );
      });
    }

    return (
      <Carousel style={styles.carousel} width={this.state.width} height={70} color={'#0071ff'}>
        {rewardsArray}
      </Carousel>
    )
  }
}