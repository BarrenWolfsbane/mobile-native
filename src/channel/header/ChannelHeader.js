import React, {
  Component
} from 'react';

import {
  Text,
  Image,
  View,
  ActivityIndicator,
  Button
} from 'react-native';

import {
  observer,
  inject
} from 'mobx-react/native'

import Icon from 'react-native-vector-icons/Ionicons';
import channelService from '../ChannelService';
import { MINDS_URI } from '../../config/Config';
import abbrev from '../../common/helpers/abbrev';

/**
 * Channel Header
 */

@observer
export default class ChannelHeader extends Component {

  /**
   * Get Channel Banner
   */
  getBannerFromChannel() {
    const channel = this.props.channel.channel;
    if (channel && channel.carousels) {
      return channel.carousels[0].src;
    }

    return `${MINDS_URI}fs/v1/banners/${channel.guid}/0/${channel.banner}/medium`;
  }

  /**
   * Get Channel Avatar
   */
  getAvatar() {
    const channel = this.props.channel.channel;
    return `${MINDS_URI}icon/${channel.guid}/large/${channel.icontime}`;
  }

  /**
   * Get Action Button, Message or Subscribe
   */
  getActionButton() {
    if(!!this.props.channel.subscribed){
      return (
        <Button
          onPress={() => { console.log('press') }}
          title="Message"
          color="#4791d6"
          accessibilityLabel="Learn more about this purple button"
        />
      );
    } else {
      return (
        <Button
          onPress={() => { this.subscribe() }}
          title="Subscribe"
          color="#4791d6"
          accessibilityLabel="Learn more about this purple button"
        />
      );
    }
  }

  subscribe() {
    let channel = this.props.channel.channel;
    this.props.channel.subscribe(channel.guid);
  }

  /**
   * Render Header
   */
  render() {

    const channel = this.props.channel.channel;
    const styles  = this.props.styles;
    const avatar  = { uri: this.getAvatar() };
    const iurl    = { uri: this.getBannerFromChannel() };

    return (
      <View>
        <Image source={iurl} style={styles.banner} resizeMode="cover" />
        <View style={styles.headertextcontainer}>
          <View style={styles.countercontainer}>
            <View style={styles.counter}>
              <Text style={styles.countertitle}>SUBSCRIBERS</Text>
              <Text style={styles.countervalue}>{abbrev(channel.subscribers_count, 0)}</Text>
            </View>
            <View style={styles.counter}>
              <Text style={styles.countertitle}>VIEWS</Text>
              <Text style={styles.countervalue}>{abbrev(channel.impressions, 0)}</Text>
            </View>
          </View>
          <View style={styles.namecontainer}>
            <View style={styles.namecol}>
              <Text style={styles.name}>{channel.name.toUpperCase()}</Text>
              <Text style={styles.username}>@{channel.username}</Text>
            </View>
            <View style={styles.buttonscol}>
              {this.getActionButton()}
              <Icon name="md-settings" size={15} />
            </View>
          </View>
          <Text style={styles.briefdescription}>{channel.briefdescription}</Text>
        </View>
        <Image source={avatar} style={styles.avatar} />
      </View>
    )
  }
}
