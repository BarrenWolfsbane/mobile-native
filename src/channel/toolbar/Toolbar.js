import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  TouchableOpacity
} from 'react-native';

import {
  observer,
} from 'mobx-react/native';

import Icon from 'react-native-vector-icons/MaterialIcons';
import IonIcon from 'react-native-vector-icons/Ionicons';
import FAIcon from 'react-native-vector-icons/FontAwesome5';

import i18n from '../../common/services/i18n.service';
import colors from '../../styles/Colors';
import featuresService from '../../common/services/features.service';

const ICON_SIZE = 22;

export default
@observer
class Toolbar extends Component {

  filterRewards = () => {
    this.props.feed.setFilter('rewards');
  }

  filterFeed = () => {
    this.props.feed.setFilter('feed');
  }

  filterImages = () => {
    this.props.feed.setFilter('images');
  }

  filterVideos = () => {
    this.props.feed.setFilter('videos');
  }

  filterBlogs = () => {
    this.props.feed.setFilter('blogs');
  }

  filterRequests = () => {
    this.props.feed.setFilter('requests');
    this.props.subscriptionRequest.load();
  }

  render() {
    const filter = this.props.feed.filter;

    const pstyles = this.props.styles;

    let rewards = null, subscriptionRequests = null;

    if (this.props.hasRewards) {
      rewards = (
        <TouchableOpacity style={styles.button} onPress={this.filterRewards}>
          <IonIcon name="ios-flash" size={ICON_SIZE} color={filter == 'rewards' ? colors.primary : color} style={styles.icon} />
          <Text style={[styles.buttontext, filter == 'rewards' ? styles.buttontextSelected : null]}>{i18n.t('rewards').toUpperCase()}</Text>
        </TouchableOpacity>
      )
    }

    if (this.props.channel.isOwner() && featuresService.has('permissions') && !this.props.channel.isOpen()) {
      subscriptionRequests = (
        <TouchableOpacity style={styles.button} onPress={this.filterRequests}>
          <FAIcon name="user-check" size={ICON_SIZE} color={filter == 'requests' ? colors.primary : color} style={styles.icon} />
          <Text style={[styles.buttontext, filter == 'requests' ? styles.buttontextSelected : null]}>{i18n.t('requests').toUpperCase()}</Text>
        </TouchableOpacity>
      )
    }

    return (
      <View style={styles.container}>
        <View style={styles.topbar}>
          <TouchableOpacity style={styles.button} onPress={this.filterFeed} testID="FeedButton">
            <Icon name="list" size={ICON_SIZE} color={filter == 'feed' ? colors.primary : color} style={styles.icon} />
            <Text style={[styles.buttontext, filter == 'feed' ? styles.buttontextSelected : null]}>{i18n.t('feed').toUpperCase()}</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button} onPress={this.filterImages} testID="ImagesButton">
            <IonIcon name="md-image" size={ICON_SIZE} color={filter == 'images' ? colors.primary : color} style={styles.icon} />
            <Text style={[styles.buttontext, filter == 'images' ? styles.buttontextSelected : null]}>{i18n.t('images').toUpperCase()}</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button} onPress={this.filterVideos} testID="VideosButton">
            <IonIcon name="md-videocam" size={ICON_SIZE} color={filter == 'videos' ? colors.primary : color} style={styles.icon} />
            <Text style={[styles.buttontext, filter == 'videos' ? styles.buttontextSelected : null]}>{i18n.t('videos').toUpperCase()}</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button} onPress={this.filterBlogs} testID="BlogsButton">
            <Icon name="subject" size={ICON_SIZE} color={filter == 'blogs' ? colors.primary : color} style={styles.icon} />
            <Text style={[styles.buttontext, filter == 'blogs' ? styles.buttontextSelected : null]}>{i18n.t('blogs.blogs').toUpperCase()}</Text>
          </TouchableOpacity>
          {rewards}
          {subscriptionRequests}
        </View>
      </View>
    );
  }
}

const color = '#444'

const styles = StyleSheet.create({
  container: {
    height: 55,
    display: 'flex',
    flexDirection: 'row',
    paddingTop: 5,
    paddingLeft: 0,
    paddingRight: 0,
    paddingBottom: 5,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: '#EEE',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: '#EEE',
  },
  icon: {
    height: 25,
  },
  topbar: {
    flex: 1,
    justifyContent: 'space-between',
    flexDirection: 'row',
  },
  buttontext: {
    paddingTop:5,
    fontSize: 10,
    color: '#444',
  },
  buttontextSelected: {
    color: colors.primary
  },
  button: {
    flex: 1,
    justifyContent: 'space-between',
    flexDirection: 'column',
    alignItems: 'center',
    padding: 3,
  },
});