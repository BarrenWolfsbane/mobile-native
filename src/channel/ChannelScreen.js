import React, {
  Component
} from 'react';

import {
  StyleSheet,
  FlatList,
  Text,
  Image,
  View,
  ScrollView,
  Button
} from 'react-native';

import {
  observer,
  inject
} from 'mobx-react/native'

import { Icon } from 'react-native-elements'
import { MINDS_URI } from '../config/Config';

import RewardsCarousel from './carousel/RewardsCarousel';
import ChannelHeader from './header/ChannelHeader';
import Toolbar from './toolbar/Toolbar';
import NewsfeedList from '../newsfeed/NewsfeedList';
import CenteredLoading from '../common/components/CenteredLoading';

/**
 * Channel Screen
 */
@inject('user')
@inject('channel')
@inject('channelfeed')
@observer
export default class ChannelScreen extends Component {

  /**
   * Disable navigation bar
   */
  static navigationOptions = {
    header: null
  }

  /**
   * Load data on mount
   */
  componentWillMount() {
    const guid = this.getGuid();
    this.props.channel.load(guid);
    this.props.channelfeed.setGuid(guid);
    this.props.channelfeed.loadFeed();
    this.props.channel.loadrewards(guid);
  }

  componentWillUnmount() {
    this.props.channelfeed.clearFeed();
    this.props.channel.clear();
  }

  getGuid(){
    return this.props.navigation.state.params.guid;
  }

  /**
   * Render
   */
  render() {
    const channel     = this.props.channel.channel;
    const rewards     = this.props.channel.rewards;
    const channelfeed = this.props.channelfeed;
    const guid        = this.getGuid();

    if (!channel.guid) {
      return (
        <CenteredLoading />
      );
    }

    let carousel = null;

    // carousel only visible if we have data
    if (rewards.merged && rewards.merged.length && channelfeed.showrewards) {
      carousel = (
        <View style={styles.carouselcontainer}>
          <RewardsCarousel rewards={rewards.merged} />
        </View>
      );
    }

    // channel header
    const header = (
      <View>
        <ChannelHeader styles={styles} me={this.props.user.me} channel={this.props.channel} navigation={this.props.navigation} />
        <Toolbar hasRewards={rewards.merged && rewards.merged.length}/>
        {carousel}
        <Icon color="white" containerStyle={styles.gobackicon} size={30} name='arrow-back' onPress={() => this.props.navigation.goBack()} />
      </View>
    );

    return (
      <NewsfeedList newsfeed={channelfeed} guid={guid} header={header} navigation={this.props.navigation} />
    );
  }
}

// style
const styles = StyleSheet.create({
  gobackicon: {
    position: 'absolute',
    left: 0,
    top: 0,
    height: 40,
    width: 40,
  },
  headertextcontainer: {
    padding: 8,
    paddingLeft: 15,
    paddingRight: 15,
    alignItems: 'stretch',
    flexDirection: 'column',
    width: '100%',
  },
  namecol: {
    flex:1
  },
  namecontainer: {
    flexDirection: 'row',
    flex:1
  },
  buttonscol: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
    width: 100,
    alignSelf:'flex-end'
  },
  carouselcontainer: {
    flex: 1,
    paddingBottom: 20
  },
  banner: {
    flexDirection: 'row',
    alignItems: 'stretch',
    width: '100%',
    height: 190,
  },
  briefdescription: {
    fontSize: 12,
    paddingTop: 25,
    paddingBottom: 20,
    color: '#919191'
  },
  headercontainer: {
    flex: 1,
    height: 200,
    flexDirection: 'row',
  },
  username: {
    fontSize: 10,
    color: '#999'
  },
  name: {
    fontWeight: 'bold',
    fontSize: 16,
  },
  countertitle: {
    color: '#666',
    fontSize: 10
  },
  countervalue: {
    paddingTop: 5,
    fontWeight: 'bold',
  },
  countercontainer: {
    paddingLeft: 130,
    height: 60,
    flexDirection: 'row'
  },
  counter: {
    flexDirection: 'column',
    alignItems: 'center',
    flex: 1
  },
  avatar: {
    position: 'absolute',
    left: 20,
    top: 135,
    height: 110,
    width: 110,
    borderRadius: 55
  },
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#FFF'
  },
});