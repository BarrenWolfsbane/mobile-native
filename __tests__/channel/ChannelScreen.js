import 'react-native';
import React from 'react';
//import { Platform, CameraRoll, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';
import ChannelScreen from '../../src/channel/ChannelScreen';
import { activitiesServiceFaker } from '../../__mocks__/fake/ActivitiesFaker';
import renderer from 'react-test-renderer';

import UserStore from '../../src/auth/UserStore';


import RewardsCarousel from '../../src/channel/carousel/RewardsCarousel';
import ChannelHeader from '../../src/channel/header/ChannelHeader';
import Toolbar from '../../src/channel/toolbar/Toolbar';
import NewsfeedList from '../../src/newsfeed/NewsfeedList';
import channelService from '../../src/channel/ChannelService';
import CenteredLoading from '../../src/common/components/CenteredLoading';
import Button from '../../src/common/components/Button';
import colors from '../../src/styles/Colors';
import BlogCard from '../../src/blogs/BlogCard';
import CaptureFab from '../../src/capture/CaptureFab';
import { CommonStyle } from '../../src/styles/Common';
import UserModel from '../../src/channel/UserModel';
import Touchable from '../../src/common/components/Touchable';
import session from '../../src/common/services/session.service';
import ChannelStore from '../../src/channel/ChannelStore';
import featuresService from '../../src/common/services/features.service';

jest.mock('../../src/common/helpers/abortableFetch');
jest.mock('../../src/channel/UserModel');
jest.mock('../../src/newsfeed/ActivityModel');
jest.mock('../../src/channel/carousel/RewardsCarousel', () => 'RewardsCarousel');
jest.mock('../../src/channel/header/ChannelHeader', () => 'ChannelHeader');
jest.mock('../../src/channel/toolbar/Toolbar', () => 'Toolbar');
jest.mock('../../src/common/components/CenteredLoading', () => 'CenteredLoading');
jest.mock('../../src/common/components/Button', () => 'Button');
jest.mock('../../src/common/components/FeedList', () => 'FeedList');
jest.mock('../../src/capture/CaptureFab', () => 'CaptureFab');
jest.mock('../../src/blogs/BlogCard', () => 'BlogCard');
jest.mock('../../src/common/components/Touchable', () => 'Touchable');
jest.mock('../../src/common/services/boosted-content.service');

/**
 * Tests
 */
describe('Channel screen component', () => {

  let store, screen, entities, channel, navigation, activityResponse, user;

  beforeEach(() => {

    let activityResponse = activitiesServiceFaker().load(1);
    let activity = activityResponse.activities[0];
    activity.wire_totals = { tokens: 20};
    activity.impressions = 20;

    navigation = { navigate: jest.fn() };
    navigation.state = {
        params: {
            entity: activity
        }
    };
    channelStore = {
      channel: {
        guid:'125',
        blocked: false,
      },
      setChannel: jest.fn(),
      feedStore: {
        setChannel: jest.fn(),
      },
      rewards: {
        merged: []
      },
      load: jest.fn(),
    };

    user = new UserStore();
    channel = { store: jest.fn() };

    channel.store.mockReturnValue(channelStore);

    screen = shallow(
      <ChannelScreen.wrappedComponent channel={channel} user={user} navigation={navigation} />
    );

    screen.update();
  });

  it('should renders correctly', () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have the expectedComponents while loading, also check newsfeed', async () => {
    screen.update();

    expect(screen.find('FeedList')).toHaveLength(1);
    let render = screen.dive();
    expect(render.find('CenteredLoading')).toHaveLength(0);
    expect(screen.find('CaptureFab')).toHaveLength(1);
  });


  it('should have the expectedComponents while loading, also check flatlist for blocked', async () => {
    screen.update();
    channelStore = {
      channel: {
        guid:'125',
        blocked: true,
      },
      setChannel: jest.fn(),
      feedStore: {
        setChannel: jest.fn(),
      },
      rewards: {
        merged: []
      },
      load: jest.fn(),
    };

    user = new UserStore();
    channel = { store: jest.fn() };

    channel.store.mockReturnValue(channelStore);

    screen = shallow(
      <ChannelScreen.wrappedComponent channel={channel} user={user} navigation={navigation} />
    );
    expect(screen.find('FlatList')).toHaveLength(1);
    let render = screen.dive();
    expect(render.find('CenteredLoading')).toHaveLength(0);
    expect(screen.find('CaptureFab')).toHaveLength(1);
  });

});