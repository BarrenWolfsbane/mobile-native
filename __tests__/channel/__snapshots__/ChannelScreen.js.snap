// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel screen component should renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <FeedList
    emptyMessage={null}
    header={
      <View>
        <ChannelHeader
          navigation={
            Object {
              "navigate": [MockFunction],
              "state": Object {
                "params": Object {
                  "entity": Object {
                    "_list": Object {
                      "viewed": Object {
                        "addViewed": [Function],
                        "viewed": Map {
                          "1019155171608096768" => true,
                        },
                      },
                    },
                    "attachment_guid": false,
                    "blurb": false,
                    "container_guid": "activityguid0",
                    "custom_data": false,
                    "custom_type": false,
                    "description": "Congratulations! ",
                    "edited": "",
                    "getThumbSource": [Function],
                    "guid": "activityguid0",
                    "impressions": 20,
                    "mature": false,
                    "message": "Message",
                    "ownerObj": Object {
                      "getAvatarSource": [Function],
                      "guid": "824853017709780997",
                      "subtype": false,
                      "time_created": "1522036284",
                      "type": "user",
                    },
                    "owner_guid": "824853017709780997",
                    "parent_guid": "838106762591510528",
                    "perma_url": false,
                    "rowKey": "something0",
                    "shouldBeBlured": [MockFunction],
                    "thumbnail_src": false,
                    "time_created": "1522036284",
                    "title": "TITLE",
                    "type": "activity",
                    "wire_totals": Object {
                      "tokens": 20,
                    },
                  },
                },
              },
            }
          }
          store={
            Object {
              "channel": Object {
                "blocked": false,
                "guid": "125",
              },
              "feedStore": Object {
                "setChannel": [MockFunction],
              },
              "load": [MockFunction],
              "rewards": Object {
                "merged": Array [],
              },
              "setChannel": [MockFunction],
            }
          }
          styles={
            Object {
              "avatar": Object {
                "borderRadius": 55,
                "height": 110,
                "left": 20,
                "position": "absolute",
                "top": 135,
                "width": 110,
              },
              "banner": Object {
                "alignItems": "stretch",
                "backgroundColor": "#eee",
                "flexDirection": "row",
                "height": 190,
                "width": "100%",
              },
              "blockText": Object {
                "fontFamily": "Roboto",
                "fontSize": 20,
                "fontWeight": "500",
                "letterSpacing": 1,
                "padding": 20,
                "paddingTop": 0,
              },
              "blockTextLink": Object {
                "color": "#4690D6",
                "fontFamily": "Roboto",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 1,
                "padding": 20,
                "paddingTop": 0,
              },
              "blockView": Object {
                "alignItems": "center",
              },
              "bluebutton": Object {
                "marginRight": 8,
              },
              "briefdescription": Object {
                "color": "#777",
                "fontFamily": "Roboto",
                "fontSize": 13,
                "paddingBottom": 20,
                "paddingTop": 25,
              },
              "briefdescriptionTextInput": Object {
                "color": "#777",
                "fontFamily": "Roboto",
                "fontSize": 13,
                "maxHeight": 100,
              },
              "briefdescriptionTextInputView": Object {
                "borderColor": "#ececec",
                "borderRadius": 3,
                "borderWidth": 1,
                "marginBottom": 20,
                "marginTop": 20,
                "padding": 8,
                "paddingTop": 3,
              },
              "buttonscol": Object {
                "alignItems": "center",
                "alignSelf": "flex-start",
                "flexDirection": "row",
                "justifyContent": "flex-start",
              },
              "carouselcontainer": Object {
                "flex": 1,
                "paddingBottom": 20,
              },
              "container": Object {
                "backgroundColor": "#FFF",
                "flex": 1,
                "flexDirection": "column",
              },
              "counter": Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
              },
              "countercontainer": Object {
                "flexDirection": "row",
                "height": 60,
                "paddingLeft": 130,
              },
              "countertitle": Object {
                "color": "#666",
                "fontSize": 10,
              },
              "countervalue": Object {
                "color": "#444",
                "fontWeight": "bold",
                "paddingTop": 5,
              },
              "gobackicon": Object {
                "left": 0,
                "position": "absolute",
                "top": 16,
              },
              "headercontainer": Object {
                "flex": 1,
                "flexDirection": "row",
                "height": 200,
              },
              "headertextcontainer": Object {
                "alignItems": "stretch",
                "flexDirection": "column",
                "padding": 8,
                "paddingLeft": 15,
                "paddingRight": 15,
                "width": "100%",
              },
              "name": Object {
                "color": "#444",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "marginRight": 8,
              },
              "nameTextInput": Object {
                "borderColor": "#ececec",
                "borderRadius": 3,
                "borderWidth": 1,
                "color": "#444",
                "fontFamily": "Roboto",
                "fontSize": 20,
                "fontWeight": "700",
                "letterSpacing": 0.5,
                "marginBottom": 4,
                "padding": 3,
              },
              "namecol": Object {
                "flex": 1,
              },
              "namecontainer": Object {
                "flex": 1,
                "flexDirection": "row",
              },
              "tapOverlayView": Object {
                "alignItems": "center",
                "backgroundColor": "#000",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "opacity": 0.65,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              "username": Object {
                "color": "#999",
                "fontSize": 14,
              },
              "wrappedAvatar": Object {
                "borderRadius": 55,
                "height": 110,
                "width": 110,
              },
              "wrappedAvatarOverlayView": Object {
                "borderRadius": 55,
              },
            }
          }
        />
        <Toolbar
          feed={
            Object {
              "setChannel": [MockFunction],
            }
          }
          hasRewards={0}
        />
        <ForwardRef(_SafeAreaView)
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "top": 16,
            }
          }
        >
          <Icon
            color="#4690D6"
            name="arrow-back"
            onPress={[Function]}
            raised={true}
            reverse={false}
            reverseColor="white"
            size={22}
            underlayColor="white"
          />
        </ForwardRef(_SafeAreaView)>
      </View>
    }
    navigation={
      Object {
        "navigate": [MockFunction],
        "state": Object {
          "params": Object {
            "entity": Object {
              "_list": Object {
                "viewed": Object {
                  "addViewed": [Function],
                  "viewed": Map {
                    "1019155171608096768" => true,
                  },
                },
              },
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "activityguid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "getThumbSource": [Function],
              "guid": "activityguid0",
              "impressions": 20,
              "mature": false,
              "message": "Message",
              "ownerObj": Object {
                "getAvatarSource": [Function],
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "rowKey": "something0",
              "shouldBeBlured": [MockFunction],
              "thumbnail_src": false,
              "time_created": "1522036284",
              "title": "TITLE",
              "type": "activity",
              "wire_totals": Object {
                "tokens": 20,
              },
            },
          },
        },
      }
    }
    renderActivity={null}
  />
  <CaptureFab
    navigation={
      Object {
        "navigate": [MockFunction],
        "state": Object {
          "params": Object {
            "entity": Object {
              "_list": Object {
                "viewed": Object {
                  "addViewed": [Function],
                  "viewed": Map {
                    "1019155171608096768" => true,
                  },
                },
              },
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "activityguid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "getThumbSource": [Function],
              "guid": "activityguid0",
              "impressions": 20,
              "mature": false,
              "message": "Message",
              "ownerObj": Object {
                "getAvatarSource": [Function],
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "rowKey": "something0",
              "shouldBeBlured": [MockFunction],
              "thumbnail_src": false,
              "time_created": "1522036284",
              "title": "TITLE",
              "type": "activity",
              "wire_totals": Object {
                "tokens": 20,
              },
            },
          },
        },
      }
    }
  />
</View>
`;
